visions/webassembly.md

```diff
@@ -0,0 +1,94 @@
+# A Vision for WebAssembly Support in Swift
+
+## Introduction
+
+WebAssembly (abbreviated [Wasm](https://webassembly.github.io/spec/core/intro/introduction.html#wasm)) is a virtual 
+machine instruction set focused on portability, security, and high performance. It is vendor-neutral, designed and
+developed by [W3C](https://w3.org). An implementation of a WebAssembly virtual machine is usually called a
+*WebAssembly runtime*, or an [*embedder*](https://webassembly.github.io/spec/core/intro/overview.html#embedder).
+
+Despite its origins in the browser, it is a general-purpose technology that has use cases in client-side and
+server-side applications and services. WebAssembly support in Swift makes the language more appealing in those settings,
+and also brings it to the browser where it previously wasn't available at all.
+
```

Comment:
I think it would be nice to say (or organize sections)
- The current state of the WebAssembly Support
- What does WebAssembly Support bring to the language
- What kind of language evolutions will be needed or what is the current missing piece
  (I'm not sure if we really need Wasm-specific language evolutions)

Reply:
`@_extern(wasm, module:, name:)` might need some SE review.

Reply:
And `@_expose(wasm)` too 😉

Reply:
The second point is the main thing I felt was missing here. The third point could probably be part of a section after goals, ie. `Language Support` or something similar. I'm not sure the current state belongs in the vision, it should certainly be part of any associated forum post though.

The second seems partially answered by portability/security, but then the sections after that seem more about describing WebAssembly (and its different parts) itself. Maybe those should go under a `WebAssembly Components` (not quite sure if that's the right wording, especially since WebAssembly Component Model is a thing) section. So we'd have eg.
* Introduction - what you have now, but expanded a little to cover what adding support brings to Swift. This could mostly just be the portability + security sections combined into the intro + anything else you want to point out.
* WebAssembly Components - this could basically just be the two WASI/component model sections you have now, but merged together and maybe reworded slightly to link them a little more.
* Goals
* Sections for any expansion on the goals, eg. the difference pieces (including language support) that need to be added

Reply:
Thank you @bnbarham, I've addressed your feedback and added the "Goals" section, limited to 5 high-level goals for now, followed by a short "Proposed Language Features" subsection.